### ISSUE TITLE
Provide Timeout function for when checking a URL

### ISSUE THREAD

**Issue by Grey007** [Xabaril/AspNetCore.Diagnostics.HealthChecks#60]
Hi, 

Is it possible to provide a timeout for when checking a URL? If nothing responds in 1 minute then consider the URL Degraded. I'm currently doing the following: 

`AddUrlGroup(new Uri("http://anotherurl/"), "Description", Microsoft.Extensions.Diagnostics.HealthChecks.HealthStatus.Degraded)`

Help will be much appreciated!

**Comment by unaizorrilla** [comment]
Hi @Grey007 

Not now, sorry! I try to add this feature soon, of course contributions are open on this repo.

**Comment by pajzo** [comment]
Hi @unaizorrilla 

Have you think about how to implement it? Because, maybe i will give it a try.

Maybe do some kind of loop for x minutes with x seconds/minutes between check before marks the check as degraded/unhealthy.

**Comment by CarlosLanderas** [comment]
I have already implemented it. Pull request pending thought. My approach has been configuring the httpclient timeout property.

**Comment by pajzo** [comment]
@CarlosLanderas Then i misunderstood the request - i thought it should check the endpoint multiple times before marking the check as unhealthy.

**Comment by CarlosLanderas** [comment]
The thing is the default timeout for the http client is 100 secs and it the service is down it can delay the heath check report. In Beatpulse we had the timeout feature integrated in the HealthChecks core but the AspNetCore healthchecks extensions will have to implement this by themselves until some kind of timeout configuration is provided by the AspNetCore library

**Comment by agcos** [comment]
@CarlosLanderas any ideas when can we expect a PR?!

**Comment by wdamron** [comment]
On a related note, the `UriHealthCheck` currently ignores the passed-in `CancellationToken` when calling `var response = await httpClient.SendAsync(requestMessage);`

(see https://github.com/Xabaril/AspNetCore.Diagnostics.HealthChecks/blob/master/src/HealthChecks.Uris/UriHealthCheck.cs#L47)

A timeout for the health-check may be specified via the following approach:

```
serviceCollection.Configure<HealthCheckPublisherOptions>(options =>
 {
    options.Timeout = TimeSpan.FromSeconds(60); // default is 30s
});
```

**Comment by CarlosLanderas** [comment]
Merged in #85. Closing issue

**Comment by unaizorrilla** [comment]
Hi @Grey007 

A timeout feature is new on 2.2.2 package for Uri healthchecks from a @wdamron PR.

You can check functional tests in order to know how is used

https://github.com/Xabaril/AspNetCore.Diagnostics.HealthChecks/blob/master/test/FunctionalTests/HealthChecks.Uris/UriHealthCheckTests.cs
